chia/util/ints.py

```diff
@@ -5,47 +5,56 @@
 
 @parse_metadata_from_name
 class int8(StructStream):
-    pass
+    MINIMUM: int8
+    MAXIMUM: int8
```

Comment:
I think in your example you would need to make 

```
class Child(Parent[Child]):
```

to represent our scenario? I would rather not mess with this generic thing here now.. Im happy to review if you come up with something in a follow up PR or if you provide some working commit for here now but for me it seems doing this for `StructStream` now will also lead to other issues in tests or other places where its used because it then needs the inner type everywhere to make mypy happy.

Reply:
Yes, I started with the self referential generic parameter and switched to the example above as i tried to observe the generalization of the situation.  The self referencing didn't seem to be an issue other than needing to be quoted perhaps, if I remember correctly.

I'll fiddle locally and see how bad it looks really applying this to this PR.

Reply:
```
chia/util/struct_stream.py:56: error: ClassVar cannot contain type variables  [misc]
```

Welp...  ok then.  Which was actually in the above output and...  yeah.